### DISCUSSION TITLE
Connection error

### DISCUSSION THREAD

**Discussion by ghost** [ecederstrand/exchangelib#1124] [General]
Two hours ago i started getting this error URL https://outlook.office365.com/EWS/Exchange.asmx returned status code 401 , but till today everything works fine. Does anybody know, if there is problem with https://outlook.office365.com/EWS/Exchange.asmx?

**Comment by ecederstrand** [comment]
This could be one of two things:

1. An intermittent error with O365. They sometimes return HTTP 401 when the server is just overloaded.
2. They finally switched off Basic Authentication. In that case, you need to use OAuth instead. See https://ecederstrand.github.io/exchangelib/#oauth-on-office-365 for a guide on doing that.
